### PR TITLE
fix: join multiple calls at the same time (AR-3267)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-3267" title="AR-3267" target="_blank">BugAR-3267</a>  I should not be able to join 2 calls in parallel on the same client
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

You can join multiple calls at the same time.

### Causes (Optional)

Not handeled.

### Solutions

Show a popup before joining a call when there is an ongoing one to ask the if he wants to proceed.
 
### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

Wire 2023-04-04 at 5_04 PM

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1631/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6165837)(6165837)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6165837)(6165837)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1631/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6171614)(6171614)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6171614)(6171614)-staging-compat.apk)

**Review by MohamadJaara — APPROVED** [review]
LGTM

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1631/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6181067)(6181067)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6181067)(6181067)-staging-compat.apk)

**Review by gongracr — COMMENTED** [review]
Wasn't this already handled when user tried to trigger a second call while actjve on a first one? Are we handling this similarly?

**Comment by ohassine** [comment]
> Wasn't this already handled when user tried to trigger a second call while actjve on a first one? Are we handling this similarly?

yes only accepting and starting a call are handled, joining by clicking on join button is not

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1631/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6182870)(6182870)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6182870)(6182870)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/calling/incoming/IncomingCallViewModel.kt [hunk 1]
```diff
@@ -115,35 +124,47 @@ class IncomingCallViewModel @Inject constructor(
...
-            )
+            }
         }
     }
 
     companion object {
-        private const val DELAY_TIME_AFTER_ENDING_CALL = 1000L
-        private val ACCEPT_CALL_DEFAULT_BACKSTACK_NODE = BackStackMode.REMOVE_CURRENT
-        private const val ACCEPT_CALL_DEFAULT_DELAY_TIME = 200L
+        const val DELAY_END_CALL = 200L
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/calling/incoming/IncomingCallViewModel.kt:168]: Here we are using `200L` for a default delay when accepting a call where as the delay time for after ending a call before was `1000L`, should we keep `200L`?

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/all/AllConversationScreen.kt [hunk 2]
```diff
@@ -111,32 +127,28 @@ fun ConversationListEmptyStateScreen() {
...
-        onOpenConversation = {},
         onEditConversation = {},
-        onOpenUserProfile = {},
-        onOpenConversationNotificationsSettings = {},
-        onJoinCall = {}
+        onOpenConversationNotificationsSettings = {}
     )
 }
+
+private const val TAG = "WireFirebaseMessagingService"
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/all/AllConversationScreen.kt]: Shouldn't this be:
```suggestion
private const val TAG = "AllConversationScreen"
```

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt [hunk 3]
```diff
@@ -422,6 +476,9 @@ class ConversationListViewModel @Inject constructor(
             homeSnackBarState.emit(HomeSnackbarState.ClearConversationContentSuccess(isGroup))
         }
     }
+    companion object {
+        const val DELAY_END_CALL = 200L
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt:480]: Here following the same question about `1000L` or `200L`

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/call/ConversationCallViewModel.kt [hunk 4]
```diff
@@ -129,4 +155,8 @@ class ConversationCallViewModel @Inject constructor(
 
     suspend fun isConferenceCallingEnabled(conversationType: Conversation.Type): ConferenceCallingResult =
         isConferenceCallingEnabled.invoke(conversationId, conversationType)
+
+    companion object {
+        const val DELAY_END_CALL = 200L
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/call/ConversationCallViewModel.kt:160]: Here following the same question about `1000L` or `200L`

**Review by alexandreferris — COMMENTED** [review]
Great work! I just have some questions 🤔

File: app/src/main/kotlin/com/wire/android/ui/calling/incoming/IncomingCallViewModel.kt (hunk 1 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/calling/incoming/IncomingCallViewModel.kt:168]: Yes it should be fine.. It's done differently now the delay is added just after ending the call

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1631/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6224229)(6224229)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6224229)(6224229)-staging-compat.apk)

**Review by alexandreferris — APPROVED** [review]

**Pull request merged**